### ISSUE TITLE
2006 MySQL server has gone away

### ISSUE THREAD

**Issue by xezpeleta** [elkarbackup/elkarbackup#331]
I'm adding this issue from @gpulido here, as someone may help him with it:

The scheduled job fails due to a MySQL timeout. Increasing the `max_allowed_packet` and `wait_timeout` didn't work. He's using ElkarBackup v1.3.1 with Docker, but I don't know if the issue is related with the infrastructure or is some code related problem.

```
[2018-10-18 02:35:57] BnvWeb.INFO: Command succeeded.  {"source":"RunJobCommand","link":"null/client/1/job/4"}  ,
,
Warning: Error while sending QUERY packet. PID=2778 in /app/elkarbackup/vendor/doctrine/dbal/lib/Doctrine/DBAL/Driver/PDOStatement.php on line 91,
,
                                                                                                                                                                                                                                                                                                    ,
  [Doctrine\DBAL\Exception\DriverException]                                                                                                                                                                                                                                                         ,
  An exception occurred while executing 'SELECT t0.id AS id_1, t0.date AS date_2, t0.runningSince AS runningSince_3, t0.priority AS priority_4, t0.state AS state_5, t0.aborted AS aborted_6, t0.data AS data_7, t0.job_id AS job_id_8 FROM Queue t0 WHERE t0.job_id = ? LIMIT 1' with params [4]:  ,
  SQLSTATE[HY000]: General error: 2006 MySQL server has gone away                                                                                                                                                                                                                                   ,
                                                                                                                                                                                                                                                                                                    ,
,
                                                                   ,
  [Doctrine\DBAL\Driver\PDOException]                              ,
  SQLSTATE[HY000]: General error: 2006 MySQL server has gone away  ,
                                                                   ,
,
                                                                   ,
  [PDOException]                                                   ,
  SQLSTATE[HY000]: General error: 2006 MySQL server has gone away  ,
                                                                   ,
,
elkarbackup:run_job [-h|--help] [-q|--quiet] [-v|vv|vvv|--verbose] [-V|--version] [--ansi] [--no-ansi] [-n|--no-interaction] [-s|--shell] [--process-isolation] [-e|--env ENV] [--no-debug] [--] <command> <job>,
,
[2018-10-18 02:35:57] BnvWeb.ERROR: Job execution failed! {"source":"StatusReport","link":"null/client/1/job/4"}  ,

```


More info: https://github.com/elkarbackup/elkarbackup-docker/issues/26#issuecomment-430891512

**Comment by xezpeleta** [comment]
I'm not able to reproduce the problem but I'd say that after a long _rsnapshot_ command, the DB connection dies. This is apparently the failing query:

https://github.com/elkarbackup/elkarbackup/blob/671e0064ed1f69965759deee0a0a4041139c7ee6/src/Binovo/ElkarBackupBundle/Command/RunJobCommand.php#L240-L243

**Comment by gpulido** [comment]
Strange, after the error when I try to execute again the tick i receive the following:
```
/app/elkarbackup# app/console elkarbackup:tick "2018-10-19 02:00"


  [Doctrine\DBAL\Exception\ConnectionException]
  An exception occured in driver: SQLSTATE[HY000] [2002] Connection refused



  [Doctrine\DBAL\Driver\PDOException]
  SQLSTATE[HY000] [2002] Connection refused



  [PDOException]
  SQLSTATE[HY000] [2002] Connection refused
```

**Comment by xezpeleta** [comment]
Hi @gpulido 

Can you try changing RunJobCommand.php file?

Here you have the diff:

https://github.com/elkarbackup/elkarbackup/compare/master...xezpeleta:fix331-dbRenewConn

**Comment by gpulido** [comment]
Hi @xezpeleta 
I had to apply the patch manually to the docker instance. Running it now. In half an hour aprox we will know if it worked. Thanks!

**Comment by gpulido** [comment]
Hi @xezpeleta 
It worked!!
```
 app/console elkarbackup:tick "2018-10-19 02:00"
[2018-10-18 15:59:57] BnvWeb.INFO: QUEUED {"source":"StatusReport","link":"null/client/1/job/2"}
[2018-10-18 15:59:57] BnvWeb.INFO: QUEUED {"source":"StatusReport","link":"null/client/1/job/4"}
[2018-10-18 15:59:58] BnvWeb.INFO: There are no PRE scripts to run for this client. {"source":"RunPreClientScriptsCommand","link":"null/client/1"}
[2018-10-18 15:59:59] BnvWeb.INFO: There are no PRE scripts to run for this job. {"source":"RunPreJobScriptsCommand","link":"null/client/1/job/2"}
[2018-10-18 16:00:00] BnvWeb.INFO: Running "/usr/bin/rsnapshot" -c "/app/tmp/rsnapshot.1_2.cfg" sync 2>&1 {"source":"RunJobCommand","link":"null/client/1/job/2"}
[2018-10-18 16:08:29] BnvWeb.INFO: Command succeeded. Number of deleted files: 61 (reg: 61) Number of regular files transferred: 109 Total file size: 98,917,749,746 bytes Total transferred file size: 9,337,852,246 bytes Literal data: 1,051,744,795 bytes Matched data: 8,286,334,719 bytes File list size: 706,536 File list generation time: 0.001 seconds File list transfer time: 0.000 seconds Total bytes sent: 1,098,626 Total bytes received: 1,054,863,085  sent 1,098,626 bytes  received 1,054,863,085 bytes  2,072,545.07 bytes/sec total size is 98,917,749,746  speedup is 93.68 {"source":"RunJobCommand","link":"null/client/1/job/2"}
[2018-10-18 16:08:29] BnvWeb.INFO: Running "/usr/bin/rsnapshot" -c "/app/tmp/rsnapshot.1_2.cfg" Daily 2>&1 {"source":"RunJobCommand","link":"null/client/1/job/2"}
[2018-10-18 16:08:35] BnvWeb.INFO: Command succeeded.  {"source":"RunJobCommand","link":"null/client/1/job/2"}
[2018-10-18 16:08:35] BnvWeb.INFO: Client "1", Job "2" du begin. {"source":"RunJobCommand","link":"null/client/1/job/2"}
[2018-10-18 16:08:46] BnvWeb.INFO: Client "1", Job "2" du end. {"source":"RunJobCommand","link":"null/client/1/job/2"}
[2018-10-18 16:08:46] BnvWeb.INFO: There are no POST scripts to run for this job. {"source":"RunPostJobScriptsCommand","link":"null/client/1/job/2"}
[2018-10-18 16:08:46] BnvWeb.INFO: OK {"source":"StatusReport","link":"null/client/1/job/2"}
[2018-10-18 16:08:47] BnvWeb.INFO: There are no PRE scripts to run for this job. {"source":"RunPreJobScriptsCommand","link":"null/client/1/job/4"}
[2018-10-18 16:08:47] BnvWeb.INFO: Running "/usr/bin/rsnapshot" -c "/app/tmp/rsnapshot.1_4.cfg" sync 2>&1 {"source":"RunJobCommand","link":"null/client/1/job/4"}
[2018-10-18 16:32:35] BnvWeb.INFO: Command succeeded. Number of deleted files: 0 Number of regular files transferred: 0 Total file size: 446,627,721,347 bytes Total transferred file size: 0 bytes Literal data: 0 bytes Matched data: 0 bytes File list size: 3,600,679 File list generation time: 0.001 seconds File list transfer time: 0.000 seconds Total bytes sent: 70,179 Total bytes received: 10,340,522  sent 70,179 bytes  received 10,340,522 bytes  7,287.86 bytes/sec total size is 446,627,721,347  speedup is 42,900.83 {"source":"RunJobCommand","link":"null/client/1/job/4"}
[2018-10-18 16:32:35] BnvWeb.INFO: Running "/usr/bin/rsnapshot" -c "/app/tmp/rsnapshot.1_4.cfg" Daily 2>&1 {"source":"RunJobCommand","link":"null/client/1/job/4"}
[2018-10-18 16:34:28] BnvWeb.INFO: Command succeeded.  {"source":"RunJobCommand","link":"null/client/1/job/4"}

Warning: PDO::query(): MySQL server has gone away in /app/elkarbackup/vendor/doctrine/dbal/lib/Doctrine/DBAL/Driver/PDOConnection.php on line 104

Warning: PDO::query(): Error reading result set's header in /app/elkarbackup/vendor/doctrine/dbal/lib/Doctrine/DBAL/Driver/PDOConnection.php on line 104
[2018-10-18 16:34:28] BnvWeb.INFO: Client "1", Job "4" du begin. {"source":"RunJobCommand","link":"null/client/1/job/4"}
[2018-10-18 16:42:20] BnvWeb.INFO: Client "1", Job "4" du end. {"source":"RunJobCommand","link":"null/client/1/job/4"}
[2018-10-18 16:42:20] BnvWeb.INFO: There are no POST scripts to run for this job. {"source":"RunPostJobScriptsCommand","link":"null/client/1/job/4"}
[2018-10-18 16:42:20] BnvWeb.INFO: OK {"source":"StatusReport","link":"null/client/1/job/4"}
[2018-10-18 16:42:20] BnvWeb.INFO: There are no POST scripts to run for this client. {"source":"RunPostClientScriptsCommand","link":"null/client/1"}
[2018-10-18 16:42:20] BnvWeb.INFO: OK {"source":"StatusReport","link":"null/client/1"}
```

**Comment by xezpeleta** [comment]
Great! So that was the problem! :)

Thanks for your help debugging the issue

**Comment by gpulido** [comment]
@xezpeleta Is the change going to be available on the docker instance soon?

**Comment by xezpeleta** [comment]
You'll have to wait until v1.3.2. Unfortunately I can't give you a date yet.

**Comment by gpulido** [comment]
Don't worry I'll just create a docker hub fork with just the patch applied on the install script.
Again thank you for the support

**Comment by carverhaines** [comment]
@xezpeleta Still getting this error using docker images :1.3.3 and :latest from dockerhub

**Comment by gpulido** [comment]
@carvehaines I still had the error but after reading a lot of documentation and docker issues, the problem was related to the way docker compose is deploying the mysql (in case you are using the docker compose aproach)
this is the db service inside the docker compose:
```
  db:
    image: mariadb:10.4
    env_file: /var/data/config/elkarbackup/elkarbackup.env
    networks:
      - internal
    volumes:
      - /etc/localtime:/etc/localtime:ro
      - /var/data/runtime/elkarbackup/db:/var/lib/mysql
    command: |
      --innodb_log_file_size=256M
      --max_allowed_packet=2G
      --wait_timeout=28800
      --connect_timeout=30
    deploy:
      endpoint_mode: dnsrr
```
Notice the endpoint_mode.
After adding that endpoint mode, I haven't receive more errors on the backup regarding this.
I hope it could help you also

**Comment by carverhaines** [comment]
That may fix it, but the correction made here has not implemented into the docker container. I verified by running the docker container w/ the new RunJobCommand.php mounted:
`
version: '3'

services:
  elkarbackup:
    image: "elkarbackup/elkarbackup:1.3.3
    restart: unless-stopped
    environment:
      SYMFONY__DATABASE__PASSWORD: "${DBPASSWORD}"
      EB_CRON: "enabled"
      TZ: "${TZ}"
      PHP_TZ: "${TZ}"
      SYMFONY__MAILER__TRANSPORT: "smtp"
      SYMFONY__MAILER__HOST: ${SMTPHOST}
      SYMFONY__MAILER__USER: ${SMTPUSER}
      SYMFONY__MAILER__PASSWORD: ${SMTPPASS}
      SYMFONY__MAILER__FROM: ${FROMEMAIL}
    volumes:
      - ${BACKUPDIR}:/app/backups
      - ./uploads:/app/uploads
      - ./sshkeys:/app/.ssh
      - ./RunJobCommand.php:/app/elkarbackup/src/Binovo/ElkarBackupBundle/Command/RunJobCommand.php ## Fixes bug 332: https://github.com/elkarbackup/elkarbackup/pull/332
    ports:
      - 3001:80

  db:
    image: mariadb:latest
    restart: unless-stopped
    environment:
      MYSQL_ROOT_PASSWORD: "${DBPASSWORD}"
      TZ: "${TZ}"
    volumes:
      - ./db:/var/lib/mysql

`

So that file has not been put in the 1.3.3 docker container in spite of the file being included in the 1.3.2 release

**Comment by carverhaines** [comment]
So, I just did a diff of RunJobCommand.php between the one included in the docker file and the 1.3.3 release on github, and they are the same, so not sure why it worked this time...
So never mind, I will keep experimenting.
